processing/src/main/java/io/druid/query/QueryRunnerFactory.java

```diff
@@ -57,7 +57,7 @@
    * @param queryRunners Individual QueryRunner objects that produce some results
    * @return a QueryRunner that, when asked, will use the ExecutorService to runt he base QueryRunners
    */
-  public QueryRunner<T> mergeRunners(ExecutorService queryExecutor, Iterable<QueryRunner<T>> queryRunners);
+  public QueryRunner<T> mergeRunners(ExecutorService queryExecutor, Iterable<QueryRunner<T>> queryRunners, QueryType query);
```

Comment:
that is a good idea, however i still don't think of exposing query as a param in mergeRunners(..) as a problem.
will change only if you guys have strong opinion here.